---
title: "Deconvolution with a matched scRNA-seq data set"
output: rmarkdown::html_vignette
description: >
  Deconvolution with matched scRNA-seq data set.
vignette: >
  %\VignetteIndexEntry{Deconvolution with a matched scRNA-seq data set}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
---
output: github_document
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Since most tumor spatial transcriptomics (ST) data do not have matched scRNA-seq data from the same sample, SpaCE does not require a malignant reference and has a built-in immune and stromal cell reference. However, SpaCE can accept a customized reference to carry out cell type deconvolution. This tutorial demonstrates how to run SpaCE with matched scRNA-seq dataset by using a pancreatic ductal adenocarcinoma (PDAC) ST data set [Moncada et al, 2020](https://www.nature.com/articles/s41587-019-0392-8){target="_blank"}.

## Create SpaCE object

To create an SpaCE object, user need to prepare three types of input data referring to a tumor ST dataset. 

1) spatial transcriptomics count data. The spatial transcriptomics count data must be in the format of matrix with gene name (row) x spot ID (column).
2) spatial location information. The spot coordinates should be in the format of matrix with spot ID (row) x  coordinates (column). This 1st and 2nd columns represent X and Y coordinates, respectively.
3) path to the H&E image file. The image path can be NA if unavailable.

``` r
library(SpaCE)

PDAC_Path <- system.file("extdata", 'oldST_PDAC', package = 'SpaCE')
load(paste0(PDAC_Path,"/st_PDAC.rda"))

# show count matrix
counts[1:6,1:5]
       10x10 10x13 10x14 10x15 10x16
A1CF       0     0     0     0     0
A2M       13     0     4     0     0
A4GALT     1     0     0     0     0
A4GNT      0     0     1     0     0
AAAS       0     0     0     0     0
AACS       0     0     0     0     0

# show coordinate matrix
spotCoordinates[1:5,]
       X  Y
10x10 10 10
10x13 10 13
10x14 10 14
10x15 10 15
10x16 10 16

# load ST data to create an SpaCE object.
SpaCE_obj <- create.SpaCE.object(
  counts=counts,
  spotCoordinates=spotCoordinates,
  imageFile=NA,
  platform = "oldST"
)

# show this object.
str(SpaCE_obj)

```

## Deconvolve cell type identities
We provide `SpaCE.deconvolution.matched.scRNAseq` to deconvolve an SpaCE object with a customized scRNA-seq data. User need to prepare three types of input data referring to the scRNA-seq dataset.

1) single cell RNA-seq (scRNA-seq) count data. The scRNA-seq count data must be in the format of matrix with gene name (row) x cell ID (column).
2) cell annotation information. This matrix should include two columns, i,e., cellID and cellType. Each row represents a single cell.
3) Hierarchical tree of cell types. This should be organized by using a list, and the name of each element are major lineages while the value of elements are the corresponding sublineages. If a major lineage does not have any sublineages, the value of this major lineage should be itself.


``` r
# load sc data
PDAC_Path <- system.file("extdata", 'oldST_PDAC', package = 'SpaCE')
load(paste0(PDAC_Path,"/sc_PDAC.rda"))

# show count matrix
sc_counts[1:6,1:5]
        c1 c2 c3 c4 c5
A1BG     0  0  0  0  0
A1CF     0  0  0  1  0
A2M      0  0  0  0  0
A2ML1    0  0  0  0  0
A3GALT2  0  0  0  0  0
A4GALT   0  0  0  0  0

# show cell annotation matrix
sc_annotation[1:6,1:5]
   cellID bio_celltype                            
c1 "c1"   "Acinar cells"                          
c2 "c2"   "Ductal - terminal ductal like"         
c3 "c3"   "Ductal - terminal ductal like"         
c4 "c4"   "Ductal - CRISP3 high/centroacinar like"
c5 "c5"   "Cancer clone A"                        
c6 "c6"   "Cancer clone A" 

# show cell type lineage tree
head(sc_lineageTree)
$Cancer
[1] "Cancer clone A" "Cancer clone B"

$Ductal
[1] "Ductal - APOL1 high/hypoxic"            "Ductal - CRISP3 high/centroacinar like"
[3] "Ductal - MHC Class II"                  "Ductal - terminal ductal like"         

$Macrophage
[1] "Macrophages A" "Macrophages B"

$mDC
[1] "mDCs A" "mDCs B"

$`Acinar cells`
[1] "Acinar cells"

$`Endocrine cells`
[1] "Endocrine cells"
```

Then, user can run `SpaCE.deconvolution.matched.scRNAseq` to carry out cell type deconvolution.

``` r
SpaCE_obj <- SpaCE.deconvolution.matched.scRNAseq(
  SpaCE_obj, 
  sc_counts=sc_counts, 
  sc_annotation=sc_annotation, 
  sc_lineageTree=sc_lineageTree, 
  coreNo=8
)

library(patchwork)
p1 <- SpaCE.visualize.deconvolution(SpaCE_obj, cellType="Cancer clone A")
p2 <- SpaCE.visualize.deconvolution(SpaCE_obj, cellType="Cancer clone B")
p3 <- SpaCE.visualize.deconvolution(SpaCE_obj, cellType="Acinar cells")
p4 <- SpaCE.visualize.deconvolution(SpaCE_obj, cellType="Ductal - CRISP3 high/centroacinar like")
(p1+p2) / (p3+p4)

``` 

<img src="img/PDAC.png" width="100%" style="border:none" />

## Extand this function module
Although our tutorials focus on analyzing ST data from tumor samples, in principle, SpaCE can be applied to any ST datasets with matched scRNA-seq data.
