<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Deconvolution with a matched scRNA-seq data set • SpaCET</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Deconvolution with a matched scRNA-seq data set">
<meta name="description" content="Deconvolution with matched scRNA-seq data set.
">
<meta property="og:description" content="Deconvolution with matched scRNA-seq data set.
">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SpaCET</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><h6 class="dropdown-header" data-toc-skip>Core</h6></li>
    <li><a class="dropdown-item" href="../articles/visium_BC.html">Cell type deconvolution and interaction analysis</a></li>
    <li><a class="dropdown-item" href="../articles/oldST_PDAC.html">Deconvolution with a matched scRNA-seq data set</a></li>
    <li><a class="dropdown-item" href="../articles/hiresST_CRC.html">Application to high resolution spatial transcriptomics data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Misc</h6></li>
    <li><a class="dropdown-item" href="../articles/GeneSetScore.html">Gene set score calculation for spatial spots</a></li>
    <li><a class="dropdown-item" href="../articles/SpatialCorrelation.html">Spatially variable genes and co-expressed ligand–receptor interactions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/data2intelligence/SpaCET/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Deconvolution with a matched scRNA-seq data set</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/data2intelligence/SpaCET/blob/HEAD/vignettes/oldST_PDAC.Rmd" class="external-link"><code>vignettes/oldST_PDAC.Rmd</code></a></small>
      <div class="d-none name"><code>oldST_PDAC.Rmd</code></div>
    </div>

    
    
<p>Since most tumor spatial transcriptomics (ST) data do not have
matched scRNA-seq data from the same sample, SpaCET does not require
malignant, stromal and immune cell reference. However, SpaCET can still
accept a customized reference to carry out cell type deconvolution. This
tutorial demonstrates how to run SpaCET with a matched scRNA-seq dataset
by using a pancreatic ductal adenocarcinoma (PDAC) ST data set from <a href="https://www.nature.com/articles/s41587-019-0392-8" target="_blank" class="external-link">Moncada et al, 2020</a>. Each spot is 100 µm in
diameter.</p>
<div class="section level2">
<h2 id="create-spacet-object">Create SpaCET object<a class="anchor" aria-label="anchor" href="#create-spacet-object"></a>
</h2>
<p>To read your ST data into R, user can create a SpaCET object by using
<code>create.SpaCET.object</code> or
<code>create.SpaCET.object.10X</code>. Specifically, if users are
analyzing an ST dataset from 10x Visium, they only need to input
“visiumPath” by using <code>create.SpaCET.object.10X</code>. Please make
sure that “visiumPath” points to the standard output folders of 10x
SpaCET Ranger, which has both “filtered_feature_bc_matrix” and “spatial”
folders.</p>
<p>Here, since the PDAC ST data set was generated from the original ST
technology, user need <code>create.SpaCET.object</code> to create a
SpaCET object by preparing four types of input data referring to a tumor
ST sample.</p>
<ol style="list-style-type: decimal">
<li>spatial transcriptomics count data. The spatial transcriptomics
count data must be in the format of matrix with gene name (row) x spot
ID (column).</li>
<li>spatial location information. The spot coordinates should be in the
format of matrix with spot ID (row) x coordinates (column). This 1st and
2nd columns represent X and Y coordinates, respectively.</li>
<li>path to the H&amp;E image file. The image path can be NA if
unavailable.</li>
<li>platform.</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/data2intelligence/SpaCET" class="external-link">SpaCET</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">oldST_Path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">'oldST_PDAC'</span>, package <span class="op">=</span> <span class="st">'SpaCET'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">oldST_Path</span>,<span class="st">"/st_PDAC.rda"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show count matrix</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co">##        s1 s2 s3 s4 s5</span></span>
<span><span class="co">## A1CF    0  0  0  0  0</span></span>
<span><span class="co">## A2M    13  0  4  0  0</span></span>
<span><span class="co">## A4GALT  1  0  0  0  0</span></span>
<span><span class="co">## A4GNT   0  0  1  0  0</span></span>
<span><span class="co">## AAAS    0  0  0  0  0</span></span>
<span><span class="co">## AACS    0  0  0  0  0</span></span>
<span></span>
<span><span class="co"># show coordinate matrix</span></span>
<span><span class="va">spotCoordinates</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">##    coordinate_x_um coordinate_y_um</span></span>
<span><span class="co">## s1            2000            2000</span></span>
<span><span class="co">## s2            2000            2600</span></span>
<span><span class="co">## s3            2000            2800</span></span>
<span><span class="co">## s4            2000            3000</span></span>
<span><span class="co">## s5            2000            3200</span></span>
<span></span>
<span><span class="co"># load ST data to create a SpaCET object.</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.SpaCET.object.html">create.SpaCET.object</a></span><span class="op">(</span></span>
<span>  counts<span class="op">=</span><span class="va">counts</span>,</span>
<span>  spotCoordinates<span class="op">=</span><span class="va">spotCoordinates</span>,</span>
<span>  imagePath<span class="op">=</span><span class="cn">NA</span>,</span>
<span>  platform <span class="op">=</span> <span class="st">"oldST"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show this object.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">SpaCET_obj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="deconvolve-cell-lineage">Deconvolve cell lineage<a class="anchor" aria-label="anchor" href="#deconvolve-cell-lineage"></a>
</h2>
<p>We provide <code>SpaCET.deconvolution.matched.scRNAseq</code> to
deconvolve a SpaCET object with a customized scRNA-seq data. User need
to prepare three types of input data referring to the matched scRNA-seq
dataset.</p>
<ol style="list-style-type: decimal">
<li>single cell RNA-seq (scRNA-seq) count data. The scRNA-seq count data
must be in the format of matrix with gene name (row) x cell ID
(column).</li>
<li>cell annotation information. This matrix should include two columns,
i,e., cellID and cellType. Each row represents a single cell. The name
of row should be same as the column cellID.</li>
<li>Hierarchical tree of cell types. This should be organized by using a
list, and the name of each element are major lineages while the value of
elements are the corresponding sublineages. If a major lineage does not
have any sublineages, the value of this major lineage should be
itself.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load sc data</span></span>
<span><span class="va">oldST_Path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">'oldST_PDAC'</span>, package <span class="op">=</span> <span class="st">'SpaCET'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">oldST_Path</span>,<span class="st">"/sc_PDAC.rda"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show count matrix</span></span>
<span><span class="va">sc_counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="co">##         c1 c2 c3 c4 c5</span></span>
<span><span class="co">## A1BG     0  0  0  0  0</span></span>
<span><span class="co">## A1CF     0  0  0  1  0</span></span>
<span><span class="co">## A2M      0  0  0  0  0</span></span>
<span><span class="co">## A2ML1    0  0  0  0  0</span></span>
<span><span class="co">## A3GALT2  0  0  0  0  0</span></span>
<span><span class="co">## A4GALT   0  0  0  0  0</span></span>
<span></span>
<span><span class="co"># show cell annotation matrix</span></span>
<span><span class="va">sc_annotation</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">##    cellID bio_celltype                            </span></span>
<span><span class="co">## c1 "c1"   "Acinar cells"                          </span></span>
<span><span class="co">## c2 "c2"   "Ductal - terminal ductal like"         </span></span>
<span><span class="co">## c3 "c3"   "Ductal - terminal ductal like"         </span></span>
<span><span class="co">## c4 "c4"   "Ductal - CRISP3 high/centroacinar like"</span></span>
<span><span class="co">## c5 "c5"   "Cancer clone A"                        </span></span>
<span><span class="co">## c6 "c6"   "Cancer clone A" </span></span>
<span></span>
<span><span class="co"># show cell type lineage tree</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sc_lineageTree</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## $Cancer</span></span>
<span><span class="co">## [1] "Cancer clone A" "Cancer clone B"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Ductal</span></span>
<span><span class="co">## [1] "Ductal - APOL1 high/hypoxic"   "Ductal - CRISP3 high/centroacinar like"</span></span>
<span><span class="co">## [3] "Ductal - MHC Class II"         "Ductal - terminal ductal like"         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Macrophage</span></span>
<span><span class="co">## [1] "Macrophages A" "Macrophages B"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $mDC</span></span>
<span><span class="co">## [1] "mDCs A" "mDCs B"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`Acinar cells`</span></span>
<span><span class="co">## [1] "Acinar cells"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`Endocrine cells`</span></span>
<span><span class="co">## [1] "Endocrine cells"</span></span></code></pre></div>
<p>Then, user can run <code>SpaCET.deconvolution.matched.scRNAseq</code>
to carry out cell type deconvolution.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># deconvolve ST data; ~2 minutes</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.deconvolution.matched.scRNAseq.html">SpaCET.deconvolution.matched.scRNAseq</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  sc_counts<span class="op">=</span><span class="va">sc_counts</span>, </span>
<span>  sc_annotation<span class="op">=</span><span class="va">sc_annotation</span>, </span>
<span>  sc_lineageTree<span class="op">=</span><span class="va">sc_lineageTree</span>, </span>
<span>  coreNo<span class="op">=</span><span class="fl">6</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Since Windows does not support parallel computation, </span></span>
<span><span class="co"># please set coreNo=1 for Windows OS.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-the-cell-type-proportion">Visualize the cell type proportion<a class="anchor" aria-label="anchor" href="#visualize-the-cell-type-proportion"></a>
</h2>
<p>We provide <code>SpaCET.visualize.spatialFeature</code> to present
the spatial distribution of cell types.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show the spatial distribution of all cell types.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"CellFraction"</span>, </span>
<span>  spatialFeatures<span class="op">=</span><span class="st">"All"</span>, </span>
<span>  sameScaleForFraction <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pointSize <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>  nrow<span class="op">=</span><span class="fl">4</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/PDAC_scatter.png" width="100%" style="border:none"></p>
<p>To show cell type composition in pie chart, user can set
<code>spatialType = "CellTypeComposition"</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the color for cell types</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">oldST_Path</span>,<span class="st">"/colors_vector.rda"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># draw scatter pie</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>     <span class="va">SpaCET_obj</span>, </span>
<span>     spatialType <span class="op">=</span> <span class="st">"CellTypeComposition"</span>, </span>
<span>     spatialFeatures <span class="op">=</span> <span class="st">"MajorLineage"</span>, </span>
<span>     colors <span class="op">=</span> <span class="va">colors_vector</span>,</span>
<span>     pointSize <span class="op">=</span> <span class="fl">0.5</span></span>
<span> <span class="op">)</span></span></code></pre></div>
<p><img src="img/PDAC_pie.png" width="52%" style="border:none"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"CellFraction"</span>,</span>
<span>  spatialFeatures <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Cancer clone A"</span>,</span>
<span>    <span class="st">"Cancer clone B"</span>,</span>
<span>    <span class="st">"Acinar cells"</span>,</span>
<span>    <span class="st">"Ductal - CRISP3 high/centroacinar like"</span><span class="op">)</span>,</span>
<span>  nrow<span class="op">=</span><span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/PDAC_prop.png" width="100%" style="border:none"></p>
<p>User can use the following code to visualize the marker gene
expression level and verify the cell type deconvolution.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Markers for cancer clone A and B, acinar cell, and centroacinar like ductal cell</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>,</span>
<span>  spatialType <span class="op">=</span> <span class="st">"GeneExpression"</span>,</span>
<span>  spatialFeatures <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TM4SF1"</span>,<span class="st">"S100A4"</span>,<span class="st">"PRSS1"</span>,<span class="st">"CRISP3"</span><span class="op">)</span>,</span>
<span>  nrow<span class="op">=</span><span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/PDAC_expr.png" width="100%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="extand-this-function-module">Extand this function module<a class="anchor" aria-label="anchor" href="#extand-this-function-module"></a>
</h2>
<p>Although our tutorials focus on analyzing ST data from tumor samples,
in principle, SpaCET can be applied to any ST datasets with matched
scRNA-seq data.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Beibei Ru.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
