<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Spatially variable genes and co-expressed ligand–receptor interactions • SpaCET</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Spatially variable genes and co-expressed ligand–receptor interactions">
<meta name="description" content="Spatially variable genes and co-expressed ligand–receptor interactions.
">
<meta property="og:description" content="Spatially variable genes and co-expressed ligand–receptor interactions.
">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SpaCET</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><h6 class="dropdown-header" data-toc-skip>Core</h6></li>
    <li><a class="dropdown-item" href="../articles/visium_BC.html">Cell type deconvolution and interaction analysis</a></li>
    <li><a class="dropdown-item" href="../articles/oldST_PDAC.html">Deconvolution with a matched scRNA-seq data set</a></li>
    <li><a class="dropdown-item" href="../articles/hiresST_CRC.html">Application to high resolution spatial transcriptomics data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Misc</h6></li>
    <li><a class="dropdown-item" href="../articles/GeneSetScore.html">Gene set score calculation for spatial spots</a></li>
    <li><a class="dropdown-item" href="../articles/SpatialCorrelation.html">Spatially variable genes and co-expressed ligand–receptor interactions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/data2intelligence/SpaCET/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Spatially variable genes and co-expressed ligand–receptor interactions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/data2intelligence/SpaCET/blob/HEAD/vignettes/SpatialCorrelation.Rmd" class="external-link"><code>vignettes/SpatialCorrelation.Rmd</code></a></small>
      <div class="d-none name"><code>SpatialCorrelation.Rmd</code></div>
    </div>

    
    
<p>In this vignette, we demonstrate the use of <code>SpaCET</code> to
identify (1) spatially variable genes, (2) spatially co-expressed
ligand–receptor interactions, and (3) pairwise co-expressed gene pairs,
all of which are primarily based on Moran’s I measures.</p>
<div class="section level2">
<h2 id="create-spacet-object">Create SpaCET object<a class="anchor" aria-label="anchor" href="#create-spacet-object"></a>
</h2>
<p>To read your ST data into R, user can create an SpaCET object by
using <code>create.SpaCET.object</code> or
<code>create.SpaCET.object.10X</code>. Specifically, if users are
analyzing an ST dataset from 10x Visium, they only need to input
“visiumPath” by using <code>create.SpaCET.object.10X</code>. Please make
sure that “visiumPath” points to the standard output folders of 10x
Space Ranger, which has both “filtered_feature_bc_matrix” and “spatial”
folders.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/data2intelligence/SpaCET" class="external-link">SpaCET</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set the path to the in-house breast cancer ST data. </span></span>
<span><span class="co"># user can set the paths to their own data.</span></span>
<span><span class="va">visiumPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"SpaCET"</span><span class="op">)</span>, <span class="st">"extdata/Visium_BC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load ST data to create an SpaCET object.</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.SpaCET.object.10X.html">create.SpaCET.object.10X</a></span><span class="op">(</span>visiumPath <span class="op">=</span> <span class="va">visiumPath</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># calculate the QC metrics</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.quality.control.html">SpaCET.quality.control</a></span><span class="op">(</span><span class="va">SpaCET_obj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="compute-weight-matrix">Compute weight matrix<a class="anchor" aria-label="anchor" href="#compute-weight-matrix"></a>
</h2>
<p>The spatial weight matrix W is calculated based on the Radial Basis
Function (RBF) kernel as follows. dij denotes the Euclidean distance
between spot i and j on spatial coordinates in the m space. We assigned
the free parameter = 100 because the distance between a spot and its 1st
layer of neighbor spots is 100 um. We set Wii = 0 to mask coexpression
effects in the same spot. We also set Wij = 0 when the distance between
spots i and j is over 200 um, i.e., beyond two layers of neighbor spots
in Visium data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">W</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calWeights.html">calWeights</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, radius<span class="op">=</span><span class="fl">200</span>, sigma<span class="op">=</span><span class="fl">100</span>, diagAsZero<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatially-variable-genes">Spatially variable genes<a class="anchor" aria-label="anchor" href="#spatially-variable-genes"></a>
</h2>
<p>Users can define a vector of genes and test whether they are
spatially variable using <code>mode="univariate"</code>. Statistical
significance is assessed by permuting spot positions. To perform a
genome-wide analysis, set <code>item = NULL</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define a gene vector</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGFB1"</span>,<span class="st">"TGFB2"</span>,<span class="st">"TGFB3"</span>,<span class="st">"TGFBR1"</span>,<span class="st">"TGFBR2"</span>,<span class="st">"TGFBR3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># run spatial autocorrelation</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.SpatialCorrelation.html">SpaCET.SpatialCorrelation</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  mode<span class="op">=</span><span class="st">"univariate"</span>, </span>
<span>  item<span class="op">=</span><span class="va">genes</span>, </span>
<span>  W<span class="op">=</span><span class="va">W</span>, </span>
<span>  nPermutation<span class="op">=</span><span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show results</span></span>
<span><span class="va">SpaCET_obj</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">SpatialCorrelation</span><span class="op">$</span><span class="va">univariate</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatially-co-expression-ligand-receptor-interaction">Spatially co-expression ligand-receptor interaction<a class="anchor" aria-label="anchor" href="#spatially-co-expression-ligand-receptor-interaction"></a>
</h2>
<p>Users can define a two-column matrix of gene pairs and test whether
they are spatially co-expressed using <code>mode="bivariate"</code>.
Statistical significance is assessed by permuting spot positions. To
perform the analysis using the in-house ligand–receptor database, set
<code>item = NULL</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define a two-column matrix of gene pairs</span></span>
<span><span class="va">genePairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGFB1"</span>,<span class="st">"TGFB1"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGFBR1"</span>,<span class="st">"TGFBR2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># run spatial autocorrelation</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.SpatialCorrelation.html">SpaCET.SpatialCorrelation</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  mode<span class="op">=</span><span class="st">"bivariate"</span>, </span>
<span>  item<span class="op">=</span><span class="va">genePairs</span>, </span>
<span>  W<span class="op">=</span><span class="va">W</span>, </span>
<span>  nPermutation<span class="op">=</span><span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show results</span></span>
<span><span class="va">SpaCET_obj</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">SpatialCorrelation</span><span class="op">$</span><span class="va">bivariate</span></span></code></pre></div>
<p>Users can use the following code to visualize the gene expression
levels of spatially co-expressed gene pairs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>,</span>
<span>  spatialType <span class="op">=</span> <span class="st">"GeneExpression"</span>,</span>
<span>  spatialFeatures <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGFB1"</span>,<span class="st">"TGFBR2"</span><span class="op">)</span>,</span>
<span>  nrow<span class="op">=</span><span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/coexpression.png" width="100%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="pairwise-co-expression-of-genome-wide-genes">Pairwise co-expression of genome-wide genes<a class="anchor" aria-label="anchor" href="#pairwise-co-expression-of-genome-wide-genes"></a>
</h2>
<p>To perform genome-wide pairwise co-expression analysis using
<code>mode="pairwise"</code>, the output is a symmetric matrix of
Moran’s I values for each gene pair.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run spatial autocorrelation</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.SpatialCorrelation.html">SpaCET.SpatialCorrelation</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, mode<span class="op">=</span><span class="st">"pairwise"</span>, W<span class="op">=</span><span class="va">W</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show results</span></span>
<span><span class="va">SpaCET_obj</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">SpatialCorrelation</span><span class="op">$</span><span class="va">pairwise</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Beibei Ru.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
