<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Cell type deconvolution and interaction analysis • SpaCET</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Cell type deconvolution and interaction analysis">
<meta name="description" content="Cell type deconvolution and interaction analysis.
">
<meta property="og:description" content="Cell type deconvolution and interaction analysis.
">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SpaCET</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><h6 class="dropdown-header" data-toc-skip>Core</h6></li>
    <li><a class="dropdown-item" href="../articles/visium_BC.html">Cell type deconvolution and interaction analysis</a></li>
    <li><a class="dropdown-item" href="../articles/oldST_PDAC.html">Deconvolution with a matched scRNA-seq data set</a></li>
    <li><a class="dropdown-item" href="../articles/hiresST_CRC.html">Application to high resolution spatial transcriptomics data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Misc</h6></li>
    <li><a class="dropdown-item" href="../articles/GeneSetScore.html">Gene set score calculation for spatial spots</a></li>
    <li><a class="dropdown-item" href="../articles/SpatialCorrelation.html">Spatially variable genes and co-expressed ligand–receptor interactions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/data2intelligence/SpaCET/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Cell type deconvolution and interaction analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/data2intelligence/SpaCET/blob/HEAD/vignettes/visium_BC.Rmd" class="external-link"><code>vignettes/visium_BC.Rmd</code></a></small>
      <div class="d-none name"><code>visium_BC.Rmd</code></div>
    </div>

    
    
<p>This tutorial demonstrates how to run SpaCET to estimate cell
identities and cell-cell interactions by using a breast cancer spatial
transcriptomics (ST) data set from <a href="https://www.10xgenomics.com/resources/datasets/human-breast-cancer-block-a-section-1-1-standard-1-0-0" target="_blank" class="external-link">10x/Visium</a>. Each spot is 55 µm in diameter covering
<a href="https://kb.10xgenomics.com/hc/en-us/articles/360035487952-How-many-cells-are-captured-in-a-single-spot" target="_blank" class="external-link">1-10 cells</a>. Before running the tutorial, make sure
that the SpaCET package and its dependencies have been installed.</p>
<div class="section level2">
<h2 id="create-spacet-object">Create SpaCET object<a class="anchor" aria-label="anchor" href="#create-spacet-object"></a>
</h2>
<p>To read your ST data into R, user can create an SpaCET object by
using <code>create.SpaCET.object</code> or
<code>create.SpaCET.object.10X</code>. Specifically, if users are
analyzing an ST dataset from 10x Visium, they only need to input
“visiumPath” by using <code>create.SpaCET.object.10X</code>. Please make
sure that “visiumPath” points to the standard output folders of 10x
Space Ranger, which includes both (1) sequencing data, i.e.,
“filtered_feature_bc_matrix.h5” file or “filtered_feature_bc_matrix”
folder, and (2) image folder “spatial”. For <span style="color:red">Visium HD</span> data, you can set
<code>visiumPath &lt;- "/Your/data/path/to/binned_outputs/square_016um/"</code>.</p>
<p>Here, we create an SpaCET object for the current breast cancer visium
sample.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/data2intelligence/SpaCET" class="external-link">SpaCET</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set the path to the in-house breast cancer ST data. </span></span>
<span><span class="co"># user can set the paths to their own data.</span></span>
<span><span class="va">visiumPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"SpaCET"</span><span class="op">)</span>, <span class="st">"extdata/Visium_BC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load ST data to create an SpaCET object.</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.SpaCET.object.10X.html">create.SpaCET.object.10X</a></span><span class="op">(</span>visiumPath <span class="op">=</span> <span class="va">visiumPath</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show this object.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">SpaCET_obj</span><span class="op">)</span></span></code></pre></div>
<p>The UMI count matrix is stored in
<code>SpaCET_obj@input$counts</code> as a sparse matrix. Each row
represents a gene whereas each column is a spot.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show this object.</span></span>
<span><span class="va">SpaCET_obj</span><span class="op">@</span><span class="va">input</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## 8 x 6 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##             50x102 59x19 14x94 47x13 73x43 61x97</span></span>
<span><span class="co">## MIR1302-2HG      .     .     .     .     .     .</span></span>
<span><span class="co">## FAM138A          .     .     .     .     .     .</span></span>
<span><span class="co">## OR4F5            .     .     .     .     .     .</span></span>
<span><span class="co">## AL627309.1       .     .     .     .     .     1</span></span>
<span><span class="co">## AL627309.3       .     .     .     .     .     .</span></span>
<span><span class="co">## AL627309.2       .     .     .     .     .     .</span></span>
<span><span class="co">## AL627309.4       .     .     .     .     .     .</span></span>
<span><span class="co">## AL732372.1       .     .     .     .     .     .</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="show-quality-control-metrics">Show quality control metrics<a class="anchor" aria-label="anchor" href="#show-quality-control-metrics"></a>
</h2>
<p>After creating the SpaCET object, you can use
<code>SpaCET.visualize.spatialFeature</code> to show both UMI and gene
counts across all ST spots, respectively. Specifically, the “UMI” option
presents the overall UMI counts within a spot while the “Gene” option
represents the number of expressed genes (i.e., genes with at least one
UMI count) in a particular spot.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter out spots with less than 100 expressed genes</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.quality.control.html">SpaCET.quality.control</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, min.genes<span class="op">=</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the QC metrics</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"QualityControl"</span>, </span>
<span>  spatialFeatures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMI"</span>,<span class="st">"Gene"</span><span class="op">)</span>,</span>
<span>  imageBg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  imageSize <span class="op">=</span> <span class="st">"CompleteImage"</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># or set imageSize = "CaptureArea" to show the capture area only.</span></span></code></pre></div>
<p><img src="img/QC.png" width="100%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="deconvolve-st-data">Deconvolve ST data<a class="anchor" aria-label="anchor" href="#deconvolve-st-data"></a>
</h2>
<p>By two stages, <code>SpaCET.deconvolution</code> would deconvolve all
mixtures of ST spots into malignant, immune, and stromal cells. SpaCET
first estimates malignant cell fraction by a gene pattern dictionary of
copy number alterations (CNA) and expression changes in common
malignancies. Subsequently, based on an in-house hierarchical cell
lineage derived from single-cell RNA-seq data sets from diverse cancer
types, a constrained regression model is used to determine immune and
stromal cell fraction, and also include an unidentifiable component to
calibrate cellular density variations across tissue regions.</p>
<p>Specifically, user need to specify the cancer type of this tumor ST
data set by using <code>cancerType</code> parameter, which is utilized
to select cancer type-specific CNA or expression signature to infer
malignant cell fraction. The code of cancer types can be found in the
following dropdown content. When correlating expression profiles of ST
spots to the cancer type-specific signature, we set the cancer
type-specific CNA signature as the first option because chromosomal
instability is widely considered one consistent feature of human tumors.
Alternatively, if no spots strongly correlate with the CNA signature,
the cancer type-specific expression signature would be activated. This
situation might result from chromosomally stable cancer cells with low
CNA. For cancer types not included in our dictionary, we created a
pan-cancer expression signature by averaging all cancer type-specific
expression signatures.</p>
<details><summary><strong>How many caner types are in the gene pattern
dictionary?</strong>
</summary><p>SpaCET includes a gene pattern dictionary of copy number alterations
(CNA) and expression changes for 30 solid tumor types. The cancer
type-specific CNA signature of a cancer type was computed by averaging
bulk tumor CNA values on gene levels across patients. For each cancer
type, the lower quartile of patients sorted by total CNA burdens was
excluded before calculating the CNA signature. The cancer type-specific
expression signature of a cancer type was generated as log2 Fold Change
of differential expression between tumor and normal samples. Several
cancer types do not have expression signatures due to a lack of adequate
normal samples (n &lt; 10 patients). The pan-cancer expression signature
was created by averaging all cancer type-specific expression
signatures.</p>
<p><img src="img/cancerDictionary.png" width="100%" style="border:none"></p>
</details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># deconvolve ST data; ~8 minutes</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.deconvolution.html">SpaCET.deconvolution</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, cancerType<span class="op">=</span><span class="st">"BRCA"</span>, coreNo<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co"># Since Windows does not support parallel computation, </span></span>
<span><span class="co"># please set coreNo=1 for Windows OS.</span></span>
<span></span>
<span><span class="co"># show the ST deconvolution results</span></span>
<span><span class="va">SpaCET_obj</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">deconvolution</span><span class="op">$</span><span class="va">propMat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">13</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span></span>
<span><span class="co">##                   50x102 59x19        14x94        47x13        73x43</span></span>
<span><span class="co">## Malignant   2.860636e-01     1 6.845966e-02 3.899756e-01 9.608802e-01</span></span>
<span><span class="co">## CAF         3.118545e-01     0 3.397067e-01 1.111980e-01 3.372692e-02</span></span>
<span><span class="co">## Endothelial 5.510895e-02     0 1.427060e-01 3.080531e-02 5.263544e-03</span></span>
<span><span class="co">## Plasma      2.213392e-02     0 1.507382e-02 1.183170e-02 9.071809e-06</span></span>
<span><span class="co">## B cell      3.885793e-03     0 9.271616e-02 1.406470e-01 1.329085e-06</span></span>
<span><span class="co">## T CD4       1.344389e-01     0 1.554305e-02 1.249414e-01 1.112392e-05</span></span>
<span><span class="co">## T CD8       7.578696e-03     0 2.514558e-07 1.379856e-03 1.123043e-06</span></span>
<span><span class="co">## NK          7.104005e-04     0 1.670019e-06 4.890387e-08 3.562557e-07</span></span>
<span><span class="co">## cDC         1.421632e-07     0 8.278023e-02 7.584295e-02 2.851146e-07</span></span>
<span><span class="co">## pDC         1.606443e-06     0 2.283754e-02 1.805671e-02 3.878344e-07</span></span>
<span><span class="co">## Macrophage  1.703304e-01     0 5.021248e-02 9.531511e-02 9.253645e-07</span></span>
<span><span class="co">## Mast        7.905067e-08     0 1.621498e-05 1.333430e-07 1.162099e-07</span></span>
<span><span class="co">## Neutrophil  1.380073e-05     0 9.528996e-07 1.167503e-08 9.908635e-05</span></span></code></pre></div>
<p>Of note, the value from deconvolution results represents the fraction
of cell types. User may find that the fraction sum of all cell types in
a spot (column) is beyond 1. This is because SpaCET outputs the cell
fractions for both major lineages and their corresponding sublineages,
e.g., T CD4 as well as Th1, Th2… .</p>
</div>
<div class="section level2">
<h2 id="visualize-the-cell-type-proportion">Visualize the cell type proportion<a class="anchor" aria-label="anchor" href="#visualize-the-cell-type-proportion"></a>
</h2>
<p>We provide <code>SpaCET.visualize.spatialFeature</code> to present
the spatial distribution of cell types.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show the spatial distribution of malignant cells and macrophages.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"CellFraction"</span>, </span>
<span>  spatialFeatures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Malignant"</span>,<span class="st">"Macrophage"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/visualizeDeconvolution.png" width="100%" style="border:none"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show the spatial distribution of all cell types.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"CellFraction"</span>, </span>
<span>  spatialFeatures<span class="op">=</span><span class="st">"All"</span>, </span>
<span>  sameScaleForFraction <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pointSize <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>  nrow<span class="op">=</span><span class="fl">5</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Since some cell types have relatively low abundance, user </span></span>
<span><span class="co"># can get a better vision by setting `sameScaleForFraction = FALSE`. </span></span></code></pre></div>
<p><img src="img/visualizeDeconvolution2.png" width="100%" style="border:none"></p>
<p>To show cell type composition in pie chart, user can set
<code>spatialType = "CellTypeComposition"</code>. If you want to crop
the specific region like (<a href="https://github.com/data2intelligence/SpaCET/issues/48" class="external-link">#48</a>),
you can set imageSize = “CustomizedArea” and also assign 4 numbers (0~1)
to <code>CustomizedAreaScale</code>, corresponding to the figure above
(x_left, x_right, y_bottom, y_top).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the color for cell types</span></span>
<span><span class="va">hiresST_Path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">'hiresST_CRC'</span>, package <span class="op">=</span> <span class="st">'SpaCET'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">hiresST_Path</span>,<span class="st">"/colors_vector.rda"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">colors_vector</span><span class="op">[</span><span class="st">"Unidentifiable"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"grey82"</span></span>
<span></span>
<span><span class="co"># draw scatter pie (left panel)</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>     <span class="va">SpaCET_obj</span>, </span>
<span>     spatialType <span class="op">=</span> <span class="st">"CellTypeComposition"</span>, </span>
<span>     spatialFeatures <span class="op">=</span> <span class="st">"MajorLineage"</span>, </span>
<span>     colors <span class="op">=</span> <span class="va">colors_vector</span>,</span>
<span>     pointSize <span class="op">=</span> <span class="fl">3</span></span>
<span> <span class="op">)</span></span>
<span> </span>
<span><span class="co"># crop scatter pie (right panel)</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>     <span class="va">SpaCET_obj</span>, </span>
<span>     spatialType <span class="op">=</span> <span class="st">"CellTypeComposition"</span>, </span>
<span>     spatialFeatures <span class="op">=</span> <span class="st">"MajorLineage"</span>, </span>
<span>     colors <span class="op">=</span> <span class="va">colors_vector</span>,</span>
<span>     pointSize <span class="op">=</span> <span class="fl">2</span>,</span>
<span>     imageBg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>     imageSize <span class="op">=</span> <span class="st">"CustomizedArea"</span>,</span>
<span>     CustomizedAreaScale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.5</span><span class="op">)</span></span>
<span> <span class="op">)</span></span></code></pre></div>
<p><img src="img/visualizeDeconvolution3.png" width="55%" style="border:none"><img src="img/visualizeDeconvolution32.png" width="40%" style="border:none"></p>
<p>In addition, you can open up an interactive visualization panel by
setting <code>interactive=TRUE</code> to browse the deconvolution
results.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>,interactive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/visualizeDeconvolution4.png" width="94%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="estimate-cell-cell-interactions">Estimate cell-cell interactions<a class="anchor" aria-label="anchor" href="#estimate-cell-cell-interactions"></a>
</h2>
<p>After decomposing cell fractions, SpaCET can infer intercellular
interactions based on cell colocalization and ligand-receptor
coexpression. Linear correlations of cell fraction between cell types
are computed across all ST spots to evaluate cell-type colocalization.
High positive correlations indicate that cell-type pairs tend to
colocalize together. To infer physical interactions, SpaCET tests the
co-expression of ligand and receptor genes within the same ST spot for
the co-localized cell-type pairs.</p>
<div class="section level3">
<h3 id="find-co-localized-cell-type-pairs">Find co-localized cell-type pairs<a class="anchor" aria-label="anchor" href="#find-co-localized-cell-type-pairs"></a>
</h3>
<p>SpaCET provides <code>SpaCET.CCI.colocalization</code> and
<code>SpaCET.visualize.colocalization</code> to compute and visualize
the co-localized cell-type pairs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the cell-cell colocalization.</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.CCI.colocalization.html">SpaCET.CCI.colocalization</a></span><span class="op">(</span><span class="va">SpaCET_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize the cell-cell colocalization.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.colocalization.html">SpaCET.visualize.colocalization</a></span><span class="op">(</span><span class="va">SpaCET_obj</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/CCC.png" width="100%" style="border:none"></p>
<p>In this breast tumor tissue, we identified several potential
colocalized cell-type pairs (see red dots in the left panel), such as
CAFs with endothelial cells and M2 macrophages. To rule out a high cell
fraction correlation caused by similar reference profiles, we compared
the correlations between cell-type fractions and between cell-type
reference profiles (see the right panel). Although the correlation of
CAF and endothelial cell fractions is high, their profile similarity is
also proportionally high. However, the similarity between CAF and M2
macrophage references was extremely low, indicating that the CAF-M2
colocalization is not simply due to profile similarity.</p>
</div>
<div class="section level3">
<h3 id="analyze-the-l-r-network-enrichment-within-st-spots">Analyze the L-R network enrichment within ST spots<a class="anchor" aria-label="anchor" href="#analyze-the-l-r-network-enrichment-within-st-spots"></a>
</h3>
<p>Cell colocalization does not directly indicate physical interaction.
Thus, we sought further evidence for cell-cell interactions by analyzing
ligand-receptor (L-R) interactions within ST spots. From a previous <a href="https://www.nature.com/articles/ncomms8866" target="_blank" class="external-link">study</a>, we collected approximately 2,500 L-R pairs as
an in-house L-R network. For each ST data set, SpaCET shuffled the L-R
interaction network by using <a href="https://bioconductor.org/packages/release/bioc/html/BiRewire.html" target="_blank" class="external-link">BiRewire</a> package to generate 1,000 randomized
networks while preserving directed degree distributions. For a spot, an
L-R network score is defined as the sum of expression products between
all L-R pairs, divided by the average random value from 1,000 randomized
networks.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the L-R network score across ST spots.</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.CCI.LRNetworkScore.html">SpaCET.CCI.LRNetworkScore</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>,coreNo<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize the L-R network score.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"LRNetworkScore"</span>, </span>
<span>  spatialFeatures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Network_Score"</span>,<span class="st">"Network_Score_pv"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/LRNetworkScore.png" width="100%" style="border:none"></p>
<p>The L-R network score at each ST spot indicates the overall
intensities of ligand-receptor interactions at each location, but not
the specific interactions between two cell types. Thus, SpaCET further
performed an enrichment analysis of L-R network scores for each
cell-type pair. For example, for the colocalization between CAF and M2
cells in the breast tumor tissue, SpaCET grouped all ST spots into four
categories: CAF-M2 colocalized, CAF or M2 dominated, and others (see the
left and middle panels in the following figure). We found that CAF-M2
colocalized spots have more substantial L-R interaction network scores
than CAF/M2-dominated spots (right panel).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ligand-Receptor analysis for a co-localized cell-type pair</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.CCI.cellTypePair.html">SpaCET.CCI.cellTypePair</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, cellTypePair<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CAF"</span>,<span class="st">"Macrophage M2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] "CAF and Macrophage M2 have potential intercellular interaction in the current tissue."</span></span>
<span></span>
<span><span class="co"># Visualize the interaction analysis of a co-localized cell-type pair.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.cellTypePair.html">SpaCET.visualize.cellTypePair</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, cellTypePair<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CAF"</span>,<span class="st">"Macrophage M2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/CAF-M2.png" width="100%" style="border:none"></p>
</div>
</div>
<div class="section level2">
<h2 id="analyze-the-tumor-immune-interface">Analyze the tumor-immune interface<a class="anchor" aria-label="anchor" href="#analyze-the-tumor-immune-interface"></a>
</h2>
<p>Interestingly, besides the interaction significance, we found an
enrichment of CAF-M2 interactions close to boundaries between tumor and
immune/stromal regions. The distance between CAF-M2 and the tumor-immune
border was calculated by averaging the distances between each CAF-M2
interaction spot and its nearest tumor border spot. We randomly selected
the same number of spots as CAF-M2 spots from the non-malignant regions
and calculated their distances to the border as the null distribution.
The result showed that CAF-M2 interaction spots are significantly closer
to the tumor-immune boundaries.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify the Tumor-Stroma Interface</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.identify.interface.html">SpaCET.identify.interface</a></span><span class="op">(</span><span class="va">SpaCET_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize the Interface</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, spatialType <span class="op">=</span> <span class="st">"Interface"</span>, spatialFeature <span class="op">=</span> <span class="st">"Interface"</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/cancerInterface1.png" width="50%" style="border:none"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Combine the interface and interaction spots</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.combine.interface.html">SpaCET.combine.interface</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, cellTypePair<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CAF"</span>,<span class="st">"Macrophage M2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize the Interface. The spatialFeature should be formated as "Interface&amp;celltype1_celltype2". Celltype 1 and 2 is in the order of alphabet.</span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, spatialType <span class="op">=</span> <span class="st">"Interface"</span>, spatialFeature <span class="op">=</span> <span class="st">"Interface&amp;CAF_Macrophage M2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/cancerInterface2.png" width="52%" style="border:none"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compute the distance of CAF-M2 to tumor border</span></span>
<span><span class="fu"><a href="../reference/SpaCET.distance.to.interface.html">SpaCET.distance.to.interface</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, cellTypePair<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CAF"</span>, <span class="st">"Macrophage M2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/cancerInterface3.png" width="60%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="explore-cancer-cell-states">Explore cancer cell states<a class="anchor" aria-label="anchor" href="#explore-cancer-cell-states"></a>
</h2>
<p>The gene expression profiles of tumor cells (i.e., cancer cell
states) are determined by both tumor cells’ genetic background and
cell-cell interactions from the surrounding environment. As an
explorative extension, SpaCET can automatically explore the spatial
distribution of different cancer cell states.</p>
<p>Based on the deconvolution results of the breast cancer dataset, we
selected the ST spots with high fractions (&gt; 0.7) of malignant cells
as tumor spots. Then, SpaCET hierarchically clustered these malignant
spots to infer different states. The Silhouette value, measuring the
similarity among the ST spots within each cluster compared to other
clusters, was used to select the optimal cluster number. In this way,
the reference profile of different cancer cell states can be computed by
averaging the expression profile of all spots in a cluster, i.e., a
cancer cell state.</p>
<p>Similar to the deconvolution of immune major lineages into
sublineages, the overall malignant cell fraction would be decomposed to
multiple cancer cell states. Thus, the sum fraction of cancer cell
states equals to the malignant cell fraction for each spot.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># further deconvolve malignant cell states</span></span>
<span><span class="va">SpaCET_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCET.deconvolution.malignant.html">SpaCET.deconvolution.malignant</a></span><span class="op">(</span><span class="va">SpaCET_obj</span>, coreNo<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show cancer cell state fraction of the first five spots</span></span>
<span><span class="va">SpaCET_obj</span><span class="op">@</span><span class="va">results</span><span class="op">$</span><span class="va">deconvolution</span><span class="op">$</span><span class="va">propMat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Malignant cell state A"</span>,<span class="st">"Malignant cell state B"</span><span class="op">)</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span></span>
<span><span class="co">##                           50x102        59x19        14x94     47x13        73x43     61x97</span></span>
<span><span class="co">## Malignant cell state A 0.2295498 9.999900e-01 6.845962e-02 0.2038680 9.608802e-01 0.6517794</span></span>
<span><span class="co">## Malignant cell state B 0.0565137 1.239573e-11 3.921715e-08 0.1861075 2.340661e-09 0.2675332</span></span>
<span></span>
<span><span class="fu"><a href="../reference/SpaCET.visualize.spatialFeature.html">SpaCET.visualize.spatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">SpaCET_obj</span>, </span>
<span>  spatialType <span class="op">=</span> <span class="st">"CellFraction"</span>, </span>
<span>  spatialFeatures<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Malignant"</span>,<span class="st">"Malignant cell state A"</span>,<span class="st">"Malignant cell state B"</span><span class="op">)</span>, </span>
<span>  nrow<span class="op">=</span><span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="img/cancerCellState.png" width="100%" style="border:none"></p>
<p>In order to annotate the identified Cancer Cell States, you can
follow this <a href="https://data2intelligence.github.io/SpaCET/articles/GeneSetScore.html#calculate-cancer-cell-state-score" class="external-link">tutorial</a>
to calculate the known cancer cell state score across spots, and then
compare them.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Beibei Ru.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
