<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Cell type deconvolution and interaction analysis without reference.
">
<title>Cell type deconvolution and interaction analysis without reference • SpaCE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Cell type deconvolution and interaction analysis without reference">
<meta property="og:description" content="Cell type deconvolution and interaction analysis without reference.
">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpaCE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/visium_BC.html">Cell type deconvolution and interaction analysis without reference</a>
    <a class="dropdown-item" href="../articles/oldST_PDAC.html">Deconvolution with a matched scRNA-seq data set</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/data2intelligence/SpaCE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Cell type deconvolution and interaction analysis without reference</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/data2intelligence/SpaCE/blob/HEAD/vignettes/visium_BC.Rmd" class="external-link"><code>vignettes/visium_BC.Rmd</code></a></small>
      <div class="d-none name"><code>visium_BC.Rmd</code></div>
    </div>

    
    
<p>This tutorial demonstrates how to run SpaCE to estimate cell
identities and cell-cell interactions by using a breast cancer spatial
transcriptomics (ST) data set from <a href="https://www.10xgenomics.com/resources/datasets/human-breast-cancer-block-a-section-1-1-standard-1-0-0" target="_blank" class="external-link">10x Visium</a>. Before running the tutorial, make sure
that the SpaCE package and its dependencies have been installed.</p>
<div class="section level2">
<h2 id="create-space-object">Create SpaCE object<a class="anchor" aria-label="anchor" href="#create-space-object"></a>
</h2>
<p>To read your ST data into R, user can create an SpaCE object by
<code>create.SpaCE.object</code>. Specifical, if user are analyzing a
Visium dataset, they only need to input “visiumPath”. Please make sure
that “visiumPath” point the standard output folders of 10x Visium, which
has both “filtered_feature_bc_matrix” and “spatial” folders.</p>
<p>The “filtered_feature_bc_matrix”`” folder includes<br>
“barcodes.tsv.gz”: spot level barcodes<br>
“features.tsv.gz”: list of genes<br>
“matrix.mtx.gz”: (sparse) matrix of counts</p>
<p>The “spatial” folder includes<br>
“tissue_positions_list.csv” : barcodes and spatial information<br>
“tissue_lowres_image.png” : hematoxylin and eosin (H&amp;E) image<br>
“scalefactors_json.json” : scaling factors for adjusting the
coordinates</p>
<p>Here, we create an SpaCE object for this breast cancer visium
dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/data2intelligence/SpaCE" class="external-link">SpaCE</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set the path to the in-house breast cancer ST data. User can set the paths to their own data.</span></span>
<span><span class="va">visiumPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">'Visium_BC'</span>, package <span class="op">=</span> <span class="st">'SpaCE'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load ST data to create an SpaCE object.</span></span>
<span><span class="va">SpaCE_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create.SpaCE.object.html">create.SpaCE.object</a></span><span class="op">(</span>visiumPath <span class="op">=</span> <span class="va">visiumPath</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show this object.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">SpaCE_obj</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="show-key-quality-control-metrics">Show key quality control metrics<a class="anchor" aria-label="anchor" href="#show-key-quality-control-metrics"></a>
</h2>
<p>After creating the SpaCE object, user can use
<code>SpaCE.visualize.metrics</code> to show both UMI and gene counts
across all ST spots, respectively.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the QC metrics.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.metrics.html">SpaCE.visualize.metrics</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, itemQC<span class="op">=</span><span class="st">"UMI"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.metrics.html">SpaCE.visualize.metrics</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, itemQC<span class="op">=</span><span class="st">"Gene"</span><span class="op">)</span></span>
<span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span></span></code></pre></div>
<p><img src="img/QC.png" width="100%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="deconvolve-st-data">Deconvolve ST data<a class="anchor" aria-label="anchor" href="#deconvolve-st-data"></a>
</h2>
<p>By two stages, <code>SpaCE.deconvolution</code> would deconvolve all
mixtures of ST spots into malignant, immune, and stromal cells. SpaCE
first estimates malignant cell faction by a gene pattern dictionary of
copy number alterations (CNA) and expression changes in common
malignancies. Subsequently, based on an in-house hierarchical cell
lineage derived from single-cell RNA-seq data sets from diverse cancer
types, a constrained regression model is used to determine immune and
stromal cell fraction, and also include an unidentifiable component to
calibrate cellular density variations across tissue regions.</p>
<p>Specifically, user need to specify the cancer type of this tumor ST
data set by using <code>cancerType</code> parameter, which aim to select
cancer type-specific CNA or expression signature to infer malignant cell
fraction. When correlating expression profiles of ST spots to the cancer
type-specific signature, we set the cancer type-specific CNA signature
as the first option because chromosomal instability is widely considered
one consistent feature of human tumors. Alternatively, if no spots
strongly correlate with the CNA signature, the cancer type-specific
expression signature would be activated. This situation might result
from chromosomally stable cancer cells with low CNA. For cancer types
not included in our dictionary, we created a pan-cancer expression
signature by averaging all cancer type-specific expression
signatures.</p>
<details><summary><strong>How many caner types are in the gene pattern
dictionary?</strong>
</summary><p>We built a gene pattern dictionary of copy number alterations (CNA)
and expression changes for 30 solid tumor types. The cancer
type-specific CNA signature of a cancer type was computed by averaging
bulk tumor CNA values on gene levels across patients. For each cancer
type, the lower quartile of patients sorted by total CNA burdens was
excluded before calculating the CNA signature. The cancer type-specific
expression signature of a cancer type was generated as log2 Fold Change
of differential expression between tumor and normal samples. Several
cancer types do not have expression signatures due to a lack of adequate
normal samples (n &lt; 10 patients). The pan-cancer expression signature
was created by averaging all cancer type-specific expression
signatures.</p>
<img src="img/cancerDictionary.png" width="100%" style="border:none"></details><div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># deconvolve ST data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>SpaCE_obj <span class="ot">&lt;-</span> <span class="fu">SpaCE.deconvolution</span>(SpaCE_obj, <span class="at">cancerType=</span><span class="st">"BRCA"</span>, <span class="at">coreNo=</span><span class="dv">8</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># show the ST deconvolution results</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>SpaCE_obj<span class="sc">@</span>results<span class="sc">$</span>deconvolution[<span class="dv">1</span><span class="sc">:</span><span class="dv">13</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>                  50x102 59x19        14x94        47x13        73x43</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>Malignant   <span class="fl">2.860636e-01</span>     <span class="dv">1</span> <span class="fl">6.845966e-02</span> <span class="fl">3.899756e-01</span> <span class="fl">9.608802e-01</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>CAF         <span class="fl">3.118545e-01</span>     <span class="dv">0</span> <span class="fl">3.397067e-01</span> <span class="fl">1.111980e-01</span> <span class="fl">3.372692e-02</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>Endothelial <span class="fl">5.510895e-02</span>     <span class="dv">0</span> <span class="fl">1.427060e-01</span> <span class="fl">3.080531e-02</span> <span class="fl">5.263544e-03</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>Plasma      <span class="fl">2.213392e-02</span>     <span class="dv">0</span> <span class="fl">1.507382e-02</span> <span class="fl">1.183170e-02</span> <span class="fl">9.071809e-06</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>B cell      <span class="fl">3.885793e-03</span>     <span class="dv">0</span> <span class="fl">9.271616e-02</span> <span class="fl">1.406470e-01</span> <span class="fl">1.329085e-06</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>T CD4       <span class="fl">1.344389e-01</span>     <span class="dv">0</span> <span class="fl">1.554305e-02</span> <span class="fl">1.249414e-01</span> <span class="fl">1.112392e-05</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>T CD8       <span class="fl">7.578696e-03</span>     <span class="dv">0</span> <span class="fl">2.514558e-07</span> <span class="fl">1.379856e-03</span> <span class="fl">1.123043e-06</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>NK          <span class="fl">7.104005e-04</span>     <span class="dv">0</span> <span class="fl">1.670019e-06</span> <span class="fl">4.890387e-08</span> <span class="fl">3.562557e-07</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>cDC         <span class="fl">1.421632e-07</span>     <span class="dv">0</span> <span class="fl">8.278023e-02</span> <span class="fl">7.584295e-02</span> <span class="fl">2.851146e-07</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>pDC         <span class="fl">1.606443e-06</span>     <span class="dv">0</span> <span class="fl">2.283754e-02</span> <span class="fl">1.805671e-02</span> <span class="fl">3.878344e-07</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>Macrophage  <span class="fl">1.703304e-01</span>     <span class="dv">0</span> <span class="fl">5.021248e-02</span> <span class="fl">9.531511e-02</span> <span class="fl">9.253645e-07</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>Mast        <span class="fl">7.905067e-08</span>     <span class="dv">0</span> <span class="fl">1.621498e-05</span> <span class="fl">1.333430e-07</span> <span class="fl">1.162099e-07</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>Neutrophil  <span class="fl">1.380073e-05</span>     <span class="dv">0</span> <span class="fl">9.528996e-07</span> <span class="fl">1.167503e-08</span> <span class="fl">9.908635e-05</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-the-cell-type-proportion">Visualize the cell type proportion<a class="anchor" aria-label="anchor" href="#visualize-the-cell-type-proportion"></a>
</h2>
<p>We provide <code>SpaCE.visualize.deconvolution</code> to show the
spatial distribution of cell type by setting the parameter
<code>cellType</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show the spatial distribution of malignant cells and macrophages.</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.deconvolution.html">SpaCE.visualize.deconvolution</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, cellType<span class="op">=</span><span class="st">"Malignant"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.deconvolution.html">SpaCE.visualize.deconvolution</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, cellType<span class="op">=</span><span class="st">"Macrophage"</span><span class="op">)</span></span>
<span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span></span></code></pre></div>
<p><img src="img/visualizeDeconvolution.png" width="100%" style="border:none"></p>
<p>By setting the parameter <code>interactive</code> as
<code>TRUE</code>, user will open up a Shiny plot to interactively
visualize each cell type and adjust the size and alpha intensity of
spots.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show the spatial distribution of malignant cells and macrophages.</span></span>
<span><span class="fu"><a href="../reference/SpaCE.visualize.deconvolution.html">SpaCE.visualize.deconvolution</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, cellType<span class="op">=</span><span class="st">"Malignant"</span>, interactive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/visualizeDeconvolutionInteractive.png" width="100%" style="border:none"></p>
</div>
<div class="section level2">
<h2 id="find-cell-cell-interaction">Find cell-cell interaction<a class="anchor" aria-label="anchor" href="#find-cell-cell-interaction"></a>
</h2>
<p>After decomposing cell fractions, SpaCE can infer intercellular
interactions based on cell colocalization and ligand-receptor
coexpression. Linear correlations of cell fraction between cell types
are computed across all ST spots to evaluate cell-type colocalization.
High positive correlations indicate that cell-type pairs tend to
colocalize together. To infer physical interactions, SpaCE tests the
co-expression of ligand and receptor genes within the same ST spot for
the co-localized cell-type pairs.</p>
<div class="section level3">
<h3 id="identify-the-co-localized-cell-type-pairs">Identify the co-localized cell-type pairs<a class="anchor" aria-label="anchor" href="#identify-the-co-localized-cell-type-pairs"></a>
</h3>
<p>In this breast tumor tissue, we identified several potential
colocalized cell-type pairs (see the left panel in the following
figure), such as CAFs with endothelial cells and M2 macrophages. To rule
out a high cell fraction correlation caused by similar reference
profiles, we compared the correlations between cell-type fractions and
between cell-type reference profiles (see the right panel). Although the
correlation of CAF and endothelial cell fractions is high, their profile
similarity is also proportionally high. However, the similarity between
CAF and M2 macrophage references was extremely low, indicating that the
CAF-M2 colocalization is not simply due to profile similarity.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the cell-cell colocalization.</span></span>
<span><span class="va">SpaCE_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.CCI.colocalization.html">SpaCE.CCI.colocalization</a></span><span class="op">(</span><span class="va">SpaCE_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize the cell-cell colocalization.</span></span>
<span><span class="fu"><a href="../reference/SpaCE.visualize.colocalization.html">SpaCE.visualize.colocalization</a></span><span class="op">(</span><span class="va">SpaCE_obj</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/CCC.png" width="100%" style="border:none"></p>
</div>
<div class="section level3">
<h3 id="analyze-the-l-r-network-enrichment-within-st-spots">Analyze the L-R network enrichment within ST spots<a class="anchor" aria-label="anchor" href="#analyze-the-l-r-network-enrichment-within-st-spots"></a>
</h3>
<p>Cell colocalization does not directly indicate physical interaction.
Thus, we sought further evidence for cell-cell interactions by analyzing
ligand-receptor (L-R) interactions within ST spots. From a previous <a href="https://www.nature.com/articles/ncomms8866" target="_blank" class="external-link">study</a>, we collected approximately 2,500 L-R pairs as
an in-house L-R network. For each ST data set, SpaCE shuffled the L-R
interaction network by using <a href="https://bioconductor.org/packages/release/bioc/html/BiRewire.html" target="_blank" class="external-link">BiRewire</a> package to generate 1,000 randomized
networks while preserving directed degree distributions. For a spot, an
L-R network score is defined as the sum of expression products between
all L-R pairs, divided by the average random value from 1,000 randomized
networks.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the L-R network score across ST spots.</span></span>
<span><span class="va">SpaCE_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.CCI.LRNetworkScore.html">SpaCE.CCI.LRNetworkScore</a></span><span class="op">(</span><span class="va">SpaCE_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize the cell-cell colocalization.</span></span>
<span><span class="fu"><a href="../reference/SpaCE.visualize.LRNetworkScore.html">SpaCE.visualize.LRNetworkScore</a></span><span class="op">(</span><span class="va">SpaCE_obj</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/LRNetworkScore.png" width="50%" style="border:none"></p>
<p>The L-R network score at each ST spot indicates the overall
intensities of ligand-receptor interactions at each location, but not
the specific interactions between two cell types. Thus, SpaCE further
performed an enrichment analysis of L-R network scores for each
cell-type pair. For example, for the colocalization between CAF and M2
cells in the breast tumor tissue, SpaCE grouped all ST spots into four
categories: CAF-M2 colocalized, CAF or M2 dominated, and others (see the
left and middle panels in the following figure). We found that CAF-M2
colocalized spots have more substantial L-R interaction network scores
than CAF/M2-dominated spots (right panel). These results lead to the
prediction of CAF-M2 interactions, but not CAF-Endothelial
interactions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the L-R network score across ST spots.</span></span>
<span><span class="fu"><a href="../reference/SpaCE.CCI.cellTypePair.html">SpaCE.CCI.cellTypePair</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>,cellTypePair<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CAF"</span>,<span class="st">"Macrophage M2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/CAF-M2.png" width="100%" style="border:none"></p>
</div>
</div>
<div class="section level2">
<h2 id="explore-cancer-cell-states">Explore cancer cell states<a class="anchor" aria-label="anchor" href="#explore-cancer-cell-states"></a>
</h2>
<p>The gene expression profiles of tumor cells (i.e., cancer cell
states) are determined by both tumor cells’ genetic background and
cell-cell interactions from the surrounding environment. As an
explorative extension, SpaCE can automatically explore the spatial
distribution of different cancer cell states.</p>
<p>Based on the deconvolution results of the breast cancer dataset, we
selected the ST spots with high fractions (&gt; 0.7) of malignant cells
as tumor spots. Then, SpaCE hierarchically clustered these malignant
spots to infer different states. The Silhouette value, measuring the
similarity among the ST spots within each cluster compared to other
clusters, was used to select the optimal cluster number.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># further deconvolve malignant cell states</span></span>
<span><span class="va">SpaCE_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.deconvolution.malignant.html">SpaCE.deconvolution.malignant</a></span><span class="op">(</span><span class="va">SpaCE_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.deconvolution.html">SpaCE.visualize.deconvolution</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, cellType<span class="op">=</span><span class="st">"Malignant"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.deconvolution.html">SpaCE.visualize.deconvolution</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, cellType<span class="op">=</span><span class="st">"Cancer cell state A"</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpaCE.visualize.deconvolution.html">SpaCE.visualize.deconvolution</a></span><span class="op">(</span><span class="va">SpaCE_obj</span>, cellType<span class="op">=</span><span class="st">"Cancer cell state B"</span><span class="op">)</span></span>
<span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span><span class="op">+</span><span class="va">p3</span></span></code></pre></div>
<p><img src="img/cancerCellState.png" width="100%" style="border:none"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Beibei Ru, Peng Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
